rig-core/src/providers/openai.rs

```diff
@@ -510,18 +510,29 @@ impl completion::CompletionModel for CompletionModel {
             json!({
                 "model": self.model,
                 "messages": full_history,
-                "temperature": completion_request.temperature,
+
             })
         } else {
             json!({
                 "model": self.model,
                 "messages": full_history,
-                "temperature": completion_request.temperature,
                 "tools": completion_request.tools.into_iter().map(ToolDefinition::from).collect::<Vec<_>>(),
                 "tool_choice": "auto",
             })
         };
 
+        // o3-mini doesn't support temperature value
+        let request = if !self.model.starts_with("o3-mini") {
```

Comment:
tbh, `temperature` is `Option` for a reason. imo, it's better to just not include `temperature` if it's `None`. In the future, I want to improve some of these request models via some structure (so we can use `skip_serializing_if: "Option::is_none"` with serde models).